src/test/mediapipeflow_test.cpp

```diff
@@ -2100,3 +2107,271 @@ INSTANTIATE_TEST_SUITE_P(
+}
+
+TEST_F(MediapipePythonNodeTest, PythonNodeNameDoesNotExist) {
+    // Must be here - does not work when added to test::SetUp
+    // Initialize Python interpreter
+    py::scoped_interpreter guard{};  // start the interpreter and keep it alive
+    py::gil_scoped_release release;  // GIL only needed in Python custom node
+    ConstructorEnabledModelManager manager;
+    std::string testPbtxt = R"(
+    input_stream: "in"
+    output_stream: "out"
+        node {
+            name: "pythonNode2"
+            calculator: "PythonBackendCalculator"
+            input_side_packet: "PYOBJECT:pyobject"
+            input_stream: "in"
+            output_stream: "out2"
+            node_options: {
+                [type.googleapis.com / mediapipe.PythonBackendCalculatorOptions]: {
+                    handler_path: "/ovms/src/test/mediapipe/python/script.py"
+                }
+            }
+        }
+    )";
+
+    ovms::MediapipeGraphConfig mgc{"mediaDummy", "", ""};
+    DummyMediapipeGraphDefinition mediapipeDummy("mediaDummy", mgc, testPbtxt);
+    mediapipeDummy.inputConfig = testPbtxt;
+    ASSERT_EQ(mediapipeDummy.validate(manager), StatusCode::OK);
+    std::unique_ptr<NodeState> nodeState = std::make_unique<NodeState>();
+    ASSERT_EQ(mediapipeDummy.getPythonNodeState("pythonNode4", nodeState), StatusCode::PYTHON_NODE_NAME_DOES_NOT_EXISTS);
+}
+
+TEST_F(MediapipePythonNodeTest, PythonNodeInitMembers) {
+    // Must be here - does not work when added to test::SetUp
+    // Initialize Python interpreter
+    py::scoped_interpreter guard{};  // start the interpreter and keep it alive
+    py::gil_scoped_release release;  // GIL only needed in Python custom node
+    ConstructorEnabledModelManager manager;
+    std::string testPbtxt = R"(
+    input_stream: "in"
+    output_stream: "out"
+        node {
+            name: "pythonNode2"
+            calculator: "PythonBackendCalculator"
+            input_side_packet: "PYOBJECT:pyobject"
+            input_stream: "in"
+            output_stream: "out2"
+            node_options: {
+                [type.googleapis.com / mediapipe.PythonBackendCalculatorOptions]: {
+                    handler_path: "/ovms/src/test/mediapipe/python/script.py"
+                }
+            }
+        }
+    )";
+
+    ovms::MediapipeGraphConfig mgc{"mediaDummy", "", ""};
+    DummyMediapipeGraphDefinition mediapipeDummy("mediaDummy", mgc, testPbtxt);
+    mediapipeDummy.inputConfig = testPbtxt;
+    ASSERT_EQ(mediapipeDummy.validate(manager), StatusCode::OK);
+    std::unique_ptr<NodeState> nodeState = std::make_unique<NodeState>();
+    ASSERT_EQ(mediapipeDummy.getPythonNodeState("pythonNode2", nodeState), StatusCode::OK);
+
+    try {
+        py::gil_scoped_acquire acquire;
+        using namespace py::literals;
+        py::module_ sys = py::module_::import("sys");
+        py::object model_instance = py::cast<py::object>(nodeState.get()->pythonNodeState);
```

Comment:
Shouldn't the state be encapsulated in a way we do not have access to underlying python data?

Reply:
I think it exposes way too much information by having pythonNodeState public. We should have getters which are required only for MP executor. And only those should be tested.

Reply:
I thin this is not required.